adagrams.rb

```diff
@@ -0,0 +1,148 @@
+  # available in letters_in_hand
+  # Making a hash from letters_in_hand with letters as keys, and counts as values
+  letters_hand_hash = {} 
+  letters_in_hand.each do |letter| #looping thru hand, counting instances of each character
+    
+    found_letter = letter
+    if letters_hand_hash.key?(found_letter)
+      letters_hand_hash[found_letter] += 1
+    else
+      letters_hand_hash[found_letter] = 1
+    end 
+  end 
+  
+  # making a hash from user_input with letters as keys and counts as values 
+  input_hash = {}
+  input_array.each do |letter|
+    count_letter = letter
+    if input_hash.key?(count_letter)
+      input_hash[count_letter] += 1
+    else
+      input_hash[count_letter] = 1
+    end
+  end 
+  
+  # comparing input_hash to hand_hash
+  input_hash.each do |letter, count|
+    if letters_hand_hash.key?(letter)
+      hand_count = letters_hand_hash[letter]
+      if hand_count >= count
+        next
+      else
+        return false
+      end 
+    end
+  end 
+  return true 
+  
+end 
+
+# Wave 3
+def score_word(word)
+  input_array = word.upcase.chars
+  
+  scores = []
+  # Calculating the score for each letter in word 
+  input_array.each do |word|
+    if word == "A" || word == "E" || word == "I" || word == "O" || word == "U" || word == "L" || word == "N" || word == "R" || word == "S" || word == "T"
+      scores.push(1)
+    elsif word == "D" || word == "G"
+      scores.push(2)
+    elsif word == "B" || word == "C" || word == "M" || word == "P"
+      scores.push(3)
+    elsif word == "F" || word == "H" || word == "V" || word == "W" || word == "Y"
+      scores.push(4)
+    elsif word == "K"
+      scores.push(5)
+    elsif word == "J" || word == "X"
+      scores.push(8)
+    elsif word == "Q" || word =="Z"
+      scores.push(10)
+    end 
+  end 
+  
+  # Calculating the sum for all the letters in word 
+  sum = 0
+  scores.each do |score|
+    sum = sum + score
+  end 
+  
+  # If the length of word is between 7-10, add 8 points to the total score 
+  if input_array.length == 7 || input_array.length == 8 || input_array.length == 9 || input_array.length == 10
+    sum = sum + 8
+  end 
+  
+  return sum
+  
+end 
+
+# Wave 4
+def highest_score_from(words)
+  # "words" is an array of strings that is provided via driver code
+  word_scores = [] #contains score for each word
+  word_lengths = [] #contains lengths for each word
+  
+  # need to either define or get rid of "word"
+  
+  words.each do |word|
+    word_letters = word.chars #convert string to array of characters
+    word_length = word_letters.length #count the length of characters 
+    word_lengths.push(word_length) #add the length of the word into word_lengths list
+    #calculate the score for each word
+    final_score = score_word(word) 
+    word_scores.push(final_score) #add the score for each word in a list
+  end 
+  
+  # determine which word has the highest score and return associated word
+  highest_score = word_scores.max
+  
```

Comment:
You currently have 2 failing tests. The code that will fix them goes here. You need to either 1) prefer the shortest word or 2) prefer the word that uses all 10 letters, depending on input.